src/org/exist/storage/NativeBroker.java

```diff
@@ -1225,6 +1237,9 @@ public void moveCollection(final Txn transaction, final Collection collection, f
         if(collection.getURI().equals(XmldbURI.ROOT_COLLECTION_URI)) {
             throw new PermissionDeniedException("Cannot move the db root collection");
         }
+        if(isSubCollection(collection, destination)) {
+            throw new PermissionDeniedException("Cannot move collection '" + collection.getURI() + "' to it child collection '"+destination.getURI()+"'.");
+        }
```

Comment:
@shabanovd Obviously not critical. If you squash your commits I will get this merged, thank you :-)